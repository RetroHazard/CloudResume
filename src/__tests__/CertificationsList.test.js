import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CertificationList from '../components/certification_list';

const sampleData = {
    Certifications: [
        {
            certification: 'Sample Certification',
            issuer: 'Sample Issuer',
            credential_id: '123456789',
            date: '2024-05-24',
            logo: 'sample_logo.png',
            links: [
                {
                    verification: [
                        {
                            website: 'https://samplewebsite.com',
                            icon: 'sample-icon',
                            display: true,
                        },
                    ],
                    credential_info: [
                        {
                            website: 'https://samplewebsite.com',
                            icon: 'sample-icon',
                            display: true,
                        },
                    ],
                },
            ],
        },
    ],
};

// Mock the DataLoader component
jest.mock('../utils/dataLoader', () => ({
    __esModule: true,
    default: ({ children }) => children(sampleData),
}));

describe('CertificationList Component', () => {
    it('renders correctly', () => {
        const { getByText } = render(<CertificationList />);
        expect(getByText('Sample Certification')).toBeInTheDocument();
        expect(getByText('Sample Issuer')).toBeInTheDocument();
        expect(getByText(/Credential ID:/)).toBeInTheDocument(); // Using a regular expression
        expect(getByText(/Date Issued:/)).toBeInTheDocument(); // Using a regular expression
    });

    it('matches snapshot', () => {
        const { container } = render(<CertificationList />);
        expect(container).toMatchSnapshot();
    });
});