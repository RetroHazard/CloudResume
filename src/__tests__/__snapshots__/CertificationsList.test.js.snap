// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificationList Component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="flex flex-col gap-3"
    >
      <div
        class="flex gap-2 justify-between w-full"
      >
        <div
          class="flex gap-4"
        >
          <img
            alt="Sample Certification Logo"
            class="hidden rounded-xl sm:block h-[6.5rem] w-[6.5rem]"
            src="sample_logo.png"
          />
          <div
            class="flex flex-col"
          >
            <h3
              class="h3 font-extrabold text-content-title mb-0"
            >
              Sample Certification
            </h3>
            <p
              class="text-content-subtitle font-semibold leading-snug mb-0.5"
            >
              Sample Issuer
            </p>
            <p
              class="font-medium text-sm text-content-date mb-0"
            >
              <strong>
                Credential ID:
              </strong>
               
              123456789
            </p>
            <p
              class="font-medium text-sm text-content-date mb-0"
            >
              <strong>
                Date Issued:
              </strong>
               
              2024-05-24
            </p>
          </div>
        </div>
        <div
          class="flex gap-3 flex-wrap sm:flex-nowrap"
          id="Sample Certification Links"
        >
          <a
            class="social-link"
            href="https://samplewebsite.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              height="1.25em"
              icon="sample-icon"
              width="1.25em"
            />
          </a>
          <a
            class="social-link"
            href="https://samplewebsite.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              height="1.25em"
              icon="sample-icon"
              width="1.25em"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
